remote-processor/RemoteProcessorServer.cpp

```diff
@@ -54,27 +53,38 @@ CRemoteProcessorServer::~CRemoteProcessorServer()
 }
 
 // State
-bool CRemoteProcessorServer::start()
+bool CRemoteProcessorServer::start(string &strError)
 {
     assert(!_bIsStarted);
 
-    // Create server socket
-    _pListeningSocket = new CListeningSocket;
+    // Create inband pipe
+    if(pipe(_aiInbandPipe) == -1) {
+        strError  = "Could not create a pipe for remote processor communication: ";
+        strError += strerror(errno);
+        return false;
+    }
 
-    if (!_pListeningSocket->listen(_uiPort)) {
+    // Create server socket
+    std::auto_ptr<CListeningSocket> pListeningSocket(new CListeningSocket);
 
-        // Remove listening socket
-        delete _pListeningSocket;
-        _pListeningSocket = NULL;
+    if (!pListeningSocket->listen(_uiPort, strError)) {
 
         return false;
     }
 
+    // Thread needs to access to the listning socket.
+    _pListeningSocket = pListeningSocket.get();
     // Create thread
-    pthread_create(&_ulThreadId, NULL, thread_func, this);
+    if(pthread_create(&_ulThreadId, NULL, thread_func, this) != 0) {
```

Comment:
code is wrong:
On success, pthread_create() returns 0; on error, it returns an error number